test/conformance/device/urDeviceGetGlobalTimestamps.cpp

```diff
@@ -26,87 +27,93 @@ T absolute_difference(T a, T b) {
-    }
+TEST_P(urDeviceGetGlobalTimestampTest, SuccessNoTimers) {
+    UUR_KNOWN_FAILURE_ON(uur::OpenCL{"Intel(R) FPGA Emulation Device"});
+
+    ASSERT_SUCCESS(urDeviceGetGlobalTimestamps(device, nullptr, nullptr));
 }
 
-TEST_F(urDeviceGetGlobalTimestampTest, SuccessSynchronizedTime) {
-    for (auto device : devices) {
-        // get the timer resolution of the device
-        size_t deviceTimerResolutionNanoSecs = 0;
-        ASSERT_SUCCESS(uur::GetDeviceProfilingTimerResolution(
-            device, deviceTimerResolutionNanoSecs));
-        size_t delayAmountNanoSecs =
-            delayTimerMultiplier * deviceTimerResolutionNanoSecs;
-
-        uint64_t deviceStartTime = 0, deviceEndTime = 0;
-        uint64_t hostStartTime = 0, hostEndTime = 0;
-        uint64_t hostOnlyStartTime = 0, hostOnlyEndTime = 0;
-
-        ASSERT_SUCCESS(urDeviceGetGlobalTimestamps(device, &deviceStartTime,
-                                                   &hostStartTime));
-        ASSERT_SUCCESS(
-            urDeviceGetGlobalTimestamps(device, nullptr, &hostOnlyStartTime));
-        ASSERT_NE(deviceStartTime, 0);
-        ASSERT_NE(hostStartTime, 0);
-        ASSERT_NE(hostOnlyStartTime, 0);
-        ASSERT_GE(hostOnlyStartTime, hostStartTime);
-
-        // wait for timers to increment
-        std::this_thread::sleep_for(
-            std::chrono::nanoseconds(delayAmountNanoSecs));
-
-        ASSERT_SUCCESS(
-            urDeviceGetGlobalTimestamps(device, &deviceEndTime, &hostEndTime));
-        ASSERT_SUCCESS(
-            urDeviceGetGlobalTimestamps(device, nullptr, &hostOnlyEndTime));
-        ASSERT_NE(deviceEndTime, 0);
-        ASSERT_NE(hostEndTime, 0);
-        ASSERT_NE(hostOnlyEndTime, 0);
-        ASSERT_GE(hostOnlyEndTime, hostEndTime);
-
-        // check that the timers have advanced
-        ASSERT_GT(deviceEndTime, deviceStartTime);
-        ASSERT_GT(hostEndTime, hostStartTime);
-        ASSERT_GT(hostOnlyEndTime, hostOnlyStartTime);
-
-        // assert that the host/devices times are synchronized to some accuracy
-        const uint64_t deviceTimeDiff = deviceEndTime - deviceStartTime;
-        const uint64_t hostTimeDiff = hostEndTime - hostStartTime;
-        const uint64_t observedDiff =
-            absolute_difference(deviceTimeDiff, hostTimeDiff);
-        const uint64_t allowedDiff = static_cast<uint64_t>(
-            std::min(deviceTimeDiff, hostTimeDiff) * allowedTimerError);
-
-        ASSERT_LE(observedDiff, allowedDiff);
-    }
+TEST_P(urDeviceGetGlobalTimestampTest, SuccessSynchronizedTime) {
+    UUR_KNOWN_FAILURE_ON(uur::CUDA{});
+    UUR_KNOWN_FAILURE_ON(uur::HIP{});
+    UUR_KNOWN_FAILURE_ON(uur::LevelZero{});
+    UUR_KNOWN_FAILURE_ON(uur::LevelZeroV2{});
+    UUR_KNOWN_FAILURE_ON(uur::NativeCPU{});
+
+    UUR_KNOWN_FAILURE_ON(uur::OpenCL{"Intel(R) FPGA Emulation Device"});
+    UUR_KNOWN_FAILURE_ON(uur::OpenCL{"Intel(R) UHD Graphics 770"});
```

Comment:
Could also be done like this:

```suggestion
    UUR_KNOWN_FAILURE_ON(
        uur::CUDA{},
        uur::HIP{},
        uur::LevelZero{},
        uur::LevelZeroV2{},
        uur::NativeCPU{},
        uur::OpenCL{
            "Intel(R) FPGA Emulation Device",
            "Intel(R) UHD Graphics 770",
        },
    );
```

Reply:
have consolidated like this wherever possible, not sure about how the formatting ends up (it doesn't allow one per line) although that will improve when we mandate a linked issue (presumably that will lengthen each adapter entry and force them onto separate lines)

Reply:
I believe its the trailing `,` on the last element that makes `clang-format` do one per line formatting.

Reply:
TBH everything looks very readable.

Reply:
no joy unfortunately, clang-format seems intent on blocking them up even with the trailing comma